crypto/threads_win.c

```diff
@@ -251,6 +251,22 @@ int CRYPTO_atomic_load(uint64_t *val, uint64_t *ret, CRYPTO_RWLOCK *lock)
 #endif
 }
 
+int CRYPTO_atomic_load_int(int *val, int *ret, CRYPTO_RWLOCK *lock)
+{
+#if (defined(NO_INTERLOCKEDOR64))
+    if (lock == NULL || !CRYPTO_THREAD_read_lock(lock))
+        return 0;
+    *ret = *val;
+    if (!CRYPTO_THREAD_unlock(lock))
+        return 0;
+
+    return 1;
+#else
+    *ret = (int)InterlockedOr((LONG volatile *)val, 0);
```

Comment:
Hmmm....is this cast from an int * to a LONG * actually valid? is sizeof(LONG) > sizeof(int)? I might expect this to cause problems.

Reply:
AFAIK on Windows (both 32 and 64 bit) int and long are the same size. Not sure if strict aliasing rules make this pointer cast invalid though.

Reply:
On Windows LONG (and long) are always int-sized. This is due to Microsoft using long everywhere in their Win32 APIs to the point that when they made Win64, they couldn't expand long to 64 bits without breaking backwards compatibility. (This model is known as 'LLP64', as opposed to UNIX's 'LP64': https://en.wikipedia.org/wiki/64-bit_computing#64-bit_data_models )

Reply:
In that case please could you add a comment here explaining why this cast is safe?